project/src/main/webapp/script.js

```diff
@@ -20,17 +20,161 @@ window.onload = function() {
 };
 
 /**
- * Adds a official election/candidate information (brief) to the homepage.
+ * Adds a brief version of the official election/candidate information to the
+ * homepage.
+ * Queries the backend database with the user input of address.
+ *
+ * Each election has the following HTML DOM structure:
+ *   <div id="election-i">
+ *     <h3>Election name</h3>
+ *     <time>Election date</time>
+ *     <div id="election-1-positions">
+ *       <ul>
+ *         <li>
+ *           <h4>Position 1</h4>
+ *           <p>Table of Candidates</p>
+ *           <table class="candidates-table">
+ *             <tr>
+ *               <td><a href="URL">Candidate Name</a></th>
+ *               <td>Party Affiliation</th>
+ *               <td>Incumbent?</th>
+ *             </tr>
+ *           </table>
+ *         </li>
+ *         <li>
+ *           <h4>Position 2...</h4>
+ *         </li>
+ *       </ul>
+ *     </div>
+ *     <hr>
+ *   </div>
  */
-function addOfficialElectionCandidateInformation() {
-  // Capture user input of location.
+async function addBriefElectionCandidateInformation() {
+  const address = location.search.substring(location.search.indexOf('=') + 1);
 
-  // Send GET request to /data with location.
+  // Send GET request to /data with address.
+  const response = await fetch(`/data?address=${address}`);
+  const dataPackage = await response.json();
+  const elections = dataPackage.electionsData;
+
+  // Add (brief version) official election/candidate information to HTML.
+  const electionsContainer = document.getElementById('elections-container');
+  electionsContainer.innerHTML = '';
+  for (let i = 0; i < elections.length; i++) {
+    const election = elections[i];
+    const electionElement = document.createElement('div');
+    electionElement.id = `election-${i + 1}`;
+    electionElement.innerHTML =
+        `<h3>${election.name}</h3>
+         <time>${election.date}</time>`;
+    // Each election corresponds to a list of positions.
+    const positions = election.positions;
+    const positionsElement = document.createElement('div');
+    positionsElement.id = electionElement.id + `-positions`;
+    const positionsList = document.createElement('ul');
+    for (let positionIndex = 0; positionIndex < positions.length;
+        positionIndex++) {
+      const position = positions[positionIndex];
+      const positionItem = document.createElement('li');
+      positionItem.innerHTML =
+          `<h4>Position ${positionIndex + 1}: ${position.name}</h4>
+           <p>Table of Candidates</p>`;
```

Comment:
Consider making a helper function that will build a table of candidates, or one that will build a single row for a candidate, so that this loop is a bit easier to read!

Reply:
Thanks for making this suggestion. I have broken the function into 5 hierarchical helper functions:

addBriefElectionCandidateInformation
--> construct one election div
-----> construct a list of positions
--------> construct one position list item
-----------> construct a table of candidates
--------------> construct one row of candidate information